apps/greencheck/tests/test_microsoft_bucket_updater.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import logging
+
+from apps.greencheck.microsoft_bucket_updater import MicrosoftBucketUpdater
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.django_db
+class TestMicrosoftBucketUpdater:
+    def test_search_dataset(self):
+        """
+        Test the searching function to track an active endpoint
+        """
+        updater = MicrosoftBucketUpdater()
+
+        # Use dataset searcher
+        dataset = updater.search_dataset()
+
+        assert len(dataset) > 0
+
+    def test_update_bucket(self):
```

Comment:
Yeah, this is going to be a bit tricky to test, as we're relying on an external API on a domain we don't own, and the presence of files there to tell us whether we make updates.

Previously I've used the `mocker` class to mock out requests to external services and in this case I think we might need to as well.

I can show you on the call this morning